Serve internal API from enduro binary

jraddaoui opened the pull request:
@djjuhasz @sevein, we discussed splitting the API into an internal and an external implementations, as a less disruptive approach to #995. However, while working on that, having two API packages, designs, clients, etc. feels like a mess and already a disruptive change. I think there is a bigger discussion to have with the team about the ingest vs storage situation. In the meantime, I think this is a less drastic solution.

Refs #995.

jraddaoui committed:
Remove enduro-internal deployment

djjuhasz reviewed (approved):
Thanks @jraddaoui — LGTM! :+1:

jraddaoui committed:
Run a second API without auth from enduro binary

Duplicate the package, storage and upload services with authentication
disabled and serve a second API from the same enduro binary as another
group actor. Allows internal communication between services without
OIDC auth and ABAC.

[skip-codecov]
